src/main/scala/com/microsoft/azure/cosmosdb/spark/schema/CosmosDBRowConverter.scala

```diff
@@ -151,6 +153,39 @@ object CosmosDBRowConverter extends RowConverter[Document]
     }
   }
 
+  def internalRowToJSONObject(internalRow: InternalRow, schema: StructType): JSONObject = {
+    var jsonObject: JSONObject = new JSONObject()
+    schema.fields.zipWithIndex.foreach({
+      case (field, i) if internalRow.isNullAt(i) => if (field.dataType == NullType) jsonObject.remove(field.name)
+      case (field, i)                    => jsonObject.put(field.name, convertInternalRowFieldToJson(internalRow.get(i, field.dataType), field.dataType))
+    })
+    jsonObject
+  }
+
+  private def convertInternalRowFieldToJson(element: Any, elementType: DataType): Any = {
```

Comment:
The difference is because I noticed for String types in InternalRow it was necessary to read the bytes by casting as UTF8String and then creating a java.lang.String - and not directly casting as java.lang.String

Reply:
would it be cleaner to make it case String => convert to UTF8
                                                case any => convertToJson()  
?

Reply:
Also internalRowToJSONObject takes the schema as an additional parameter since InternalRow doesn't contain the schema

Reply:
Have refactored the inner function to prevent code duplication now

Reply:
great :)